homeassistant/components/green_planet_energy/sensor.py

```diff
@@ -0,0 +1,366 @@
+"""Green Planet Energy sensor platform."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+from typing import Any
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CURRENCY_EURO, UnitOfEnergy
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import dt as dt_util
+
+from .const import SENSOR_HOURS
+from .coordinator import GreenPlanetEnergyUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class GreenPlanetEnergySensorEntityDescription(SensorEntityDescription):
+    """Describes Green Planet Energy sensor entity."""
+
+    hour: int
+
+
+SENSOR_HOURLY_DESCRIPTIONS: list[GreenPlanetEnergySensorEntityDescription] = [
+    GreenPlanetEnergySensorEntityDescription(
+        key=f"gpe_price_{hour:02d}",
+        translation_key=f"price_{hour:02d}",
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        device_class=SensorDeviceClass.MONETARY,
+        suggested_display_precision=4,
+        hour=hour,
+    )
+    for hour in SENSOR_HOURS
+]
```

Comment:
I am not sure how 24 entities are, wouldn't a service in the end be more user friendly?